cas-client-core/src/main/java/org/jasig/cas/client/session/SingleSignOutHandler.java

```diff
@@ -298,18 +295,24 @@ private void destroySession(final HttpServletRequest request) {
 
         final String token = XmlUtils.getTextForElement(logoutMessage, "SessionIndex");
         if (CommonUtils.isNotBlank(token)) {
-            final HttpSession session = this.sessionMappingStorage.removeSessionByMappingId(token);
+            final ClientSession session = this.sessionMappingStorage.removeSessionByMappingId(token);
 
             if (session != null) {
                 final String sessionID = session.getId();
                 logger.debug("Invalidating session [{}] for token [{}]", sessionID, token);
 
                 try {
                     session.invalidate();
-                } catch (final IllegalStateException e) {
+                } catch (final RuntimeException e) {
                     logger.debug("Error invalidating session.", e);
                 }
-                this.logoutStrategy.logout(request);
+
+                try {
+                    request.logout();
```

Comment:
I think (unless I'm missing something) that this only throws if a Servlet 3.0 request throws `ServletException`. The strategy for logout that was previously embedded in the handler has moved to `DelegatingHttpRequest`, and it implements `HttpRequest.logout` in the same way as the previous implementation -- i.e. `logout` is a no-op for Servlet 2.5.